src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,107 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String specifiedCharacter = ",";
 
     public void getStatistic(String fromFileName, String toFileName) {
 
+        File fileFrom = new File(fromFileName);
+
+        Report reportObject = processFile(fileFrom);
+
+        String report = createReport(
+                reportObject.getTotalSupplyAmount(), reportObject.getTotalBuyAmount());
+
+        writeToFile(toFileName, report);
+
```

Comment:
You create very interesting logic, but it is hard to understand) Lets create instead of it little different methods:
readFile( String fromFileName) 
createReport( String data) 
writeToFile(toFileName, report)

Lets delete class Report, try to do this statistic without it

Reply:
In the approach suggested by you we end up with two iterations, thus, time and memory complexity is increased twise, from O(n) to O(2n), if this is fine I will fix my code to achieve better readibility as a trade off. Btw, I replaced the Report class with an array

Reply:
O(n) == O(2n) because we dont use numbers in compexity